src/main/java/core/basesyntax/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.dao.FruitDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationHandler;
+import core.basesyntax.service.ShopService;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final FruitDao fruitDao;
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(FruitDao fruitDao, OperationStrategy operationStrategy) {
+        this.fruitDao = fruitDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions) {
+        for (FruitTransaction transaction : transactions) {
+            Fruit fruit = fruitDao.getFruitByName(transaction.getFruit());
+            OperationHandler handler = operationStrategy.get(transaction.getOperation());
+            int balance = handler.executeOperation(fruit.getQuantity(), transaction.getQuantity());
```

Comment:
Be careful with the executeOperation method. According to the checklist, you need to ensure that the balance is positive after this operation when handling a Purchase. If the balance is negative, you should throw a RuntimeException.

Reply:
++